universal-application-tool-0.0.1/app/services/question/NumberQuestionDefinition.java

```diff
@@ -0,0 +1,135 @@
+      ImmutableMap<Locale, String> questionHelpText,
+      NumberQuestionDefinition.NumberValidationPredicates validationPredicates) {
+    super(version, name, path, description, questionText, questionHelpText, validationPredicates);
+  }
+
+  public NumberQuestionDefinition(
+      long version,
+      String name,
+      Path path,
+      String description,
+      ImmutableMap<Locale, String> questionText,
+      ImmutableMap<Locale, String> questionHelpText) {
+    super(
+        version,
+        name,
+        path,
+        description,
+        questionText,
+        questionHelpText,
+        NumberQuestionDefinition.NumberValidationPredicates.create());
+  }
+
+  @JsonDeserialize(
+      builder = AutoValue_NumberQuestionDefinition_NumberValidationPredicates.Builder.class)
+  @AutoValue
+  public abstract static class NumberValidationPredicates extends ValidationPredicates {
+
+    public static NumberQuestionDefinition.NumberValidationPredicates parse(String jsonString) {
+      try {
+        return mapper.readValue(
+            jsonString, AutoValue_NumberQuestionDefinition_NumberValidationPredicates.class);
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    public static NumberQuestionDefinition.NumberValidationPredicates create() {
+      return builder().build();
+    }
+
+    public static NumberQuestionDefinition.NumberValidationPredicates create(int min, int max) {
+      return builder().setMin(min).setMax(max).build();
+    }
+
+    @JsonProperty("min")
+    public abstract OptionalLong min();
+
+    @JsonProperty("max")
+    public abstract OptionalLong max();
+
+    public static NumberQuestionDefinition.NumberValidationPredicates.Builder builder() {
+      return new AutoValue_NumberQuestionDefinition_NumberValidationPredicates.Builder();
+    }
+
+    @AutoValue.Builder
+    public abstract static class Builder {
+
+      @JsonProperty("min")
+      public abstract NumberQuestionDefinition.NumberValidationPredicates.Builder setMin(
+          OptionalLong min);
+
+      public abstract NumberQuestionDefinition.NumberValidationPredicates.Builder setMin(int min);
+
+      @JsonProperty("max")
+      public abstract NumberQuestionDefinition.NumberValidationPredicates.Builder setMax(
+          OptionalLong max);
+
+      public abstract NumberQuestionDefinition.NumberValidationPredicates.Builder setMax(int max);
+
+      public abstract NumberQuestionDefinition.NumberValidationPredicates build();
+    }
+  }
+
+  public NumberQuestionDefinition.NumberValidationPredicates getNumberValidationPredicates() {
+    return (NumberQuestionDefinition.NumberValidationPredicates) getValidationPredicates();
+  }
+
+  @Override
+  public QuestionType getQuestionType() {
+    return QuestionType.NUMBER;
+  }
+
+  @Override
+  public ImmutableMap<Path, ScalarType> getScalars() {
+    return ImmutableMap.of(getNumberPath(), getNumberType());
```

Comment:
Question(blocking): Do we need to include the metadata?

Reply:
Oof! Great catch - I wonder if we can determine a way to make this less forgettable

Reply:
I could image having a protected method for those fields and call the method here? Alternatively, we could have a public base method that call all overridden methods (with a different name and make them protected) instead.